lib/broccoli/ember-app.js

```diff
@@ -1845,9 +1852,28 @@ function addOutputFile(container, assetPath, options) {
     container[outputFile] = [];
   }
 
+  if (!allowImport(container[outputFile], assetPath, options)) { return; }
+
   if (options.prepend) {
     container[outputFile].unshift(assetPath);
   } else {
     container[outputFile].push(assetPath);
   }
 }
+
+function allowImport(fileList, assetPath, options) {
+  if (fileList.indexOf(assetPath) !== -1) {
+    logger.info('Highlander Rule: duplicate `app.import(' + assetPath + ')`. Only including the last by order.');
+
+    if (options.prepend) {
```

Comment:
Order does matter. But the last one always wins. Given `[1,2,3]` and adding `{ 3, prepend: true }` the _existing_ 3 is the one which would be last in the array, and thus the one which would apply in the end.